File: school/cpp/ege/27/6057/main.py
def read_data(file: str) -> list[tuple[int, int, int]]:
    with open(file) as f:
        n = int(f.readline())
        res = []
        for _ in range(n):
            a, b, c = sorted(map(int, f.readline().split()))
            res.append((a, b, c))
        return res


def soluton(data: list[tuple[int, int, int]]) -> int:
    ans = 0
    dif = 100**100
    for triple in data:
        ans += triple[2]
        assert triple[2] == max(triple)
        if (now_diff := triple[2] - triple[1]) % 91 != 0:
            dif = min(dif, now_diff)
        elif (now_diff := triple[2] - triple[0]) % 91 != 0:
            dif = min(dif, now_diff)
    if ans % 91 == 0:
        print(f'{dif=}')
        return ans - dif
    return ans


if __name__ == '__main__':
    print(soluton(read_data('27_A_6057.txt')), end=" ")
    print(soluton(read_data('27_B_6057.txt')))

File: school/cpp/ege/27/6057/main.cpp
#include <vector>
#include <fstream>
#include <string>
#include <iostream>
#include <algorithm>

std::vector<std::vector<int>> read_data(const std::string& filename)
{
    size_t n;
    std::fstream file (filename);
    file >> n;
    std::vector<std::vector<int>> data (n, std::vector<int>(3));
    for (size_t i = 0; i != n; ++i)
        for (size_t j = 0; j != 3; ++j)
            file >> data.at(i).at(j);
    return data;
}

int solution(const std::vector<std::vector<int>>& data)
{
    int ans = 0;
    int min_diff = 10000000;
    for (const auto& triple : data)
    {
        int a, b, c;
        a = std::min(std::min(triple.at(0), triple.at(1)), triple.at(2));
        c = std::max(std::max(triple.at(0), triple.at(1)), triple.at(2));
        b = triple.at(0) + triple.at(1) + triple.at(2) - a - c;

        ans += c;
        if ((c - b) % 91 != 0)
            min_diff = std::min(min_diff, c - b);
        if ((c - a) % 91 != 0)
            min_diff = std::min(min_diff, c - a);
    }
    if (ans % 91 == 0)
        ans -= min_diff;
    return ans;
}

int main()
{
    std::cout << solution(read_data("27_A_6057.txt")) << " " << solution(read_data("27_B_6057.txt")) << std::endl;
    return 0;
}
